### PR TITLE
sqlgen: pass context to sql

### PR DESCRIPTION
**Pull request by changpingc**

### CODE REVIEW

File: sqlgen/integration_test.go [hunk 1]
```diff
@@ -121,3 +121,41 @@ func TestIntegrationBasic(t *testing.T) {
...
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+
+	// When we cancel the context after rows.Next() returns true,
+	// database/sql.(*Rows).initContextClose monitors the context
+	// and closes rows asynchronously, and subsequent rows.Scan()
+	// returns errors.New("sql: Rows are closed") instead of
+	// context.Canceled.
```
[the remark this answers is not in the document]
Comment (stephen) [reply sqlgen/integration_test.go]: maybe add in the commit message that we need to be aware of this

[the remark this answers is not in the document]
Comment (changpingc) [reply sqlgen/integration_test.go]: done.

File: sqlgen/db.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ func NewDB(conn *sql.DB, schema *Schema) *DB {
 			clause, args = selectQuery.ToSQL()
 
 			// Then, run the SQL query.
-			res, err := db.Conn.Query(clause, args...)
+			res, err := db.Conn.QueryContext(ctx, clause, args...)
```
[the remark this answers is not in the document]
Comment (stephen) [reply sqlgen/db.go]: i.e. many things will call `Invoke(ctx)`, but only one of their contexts will be used.

[the remark this answers is not in the document]
Comment (changpingc) [reply sqlgen/db.go]: oh right not much we can do there.

**Pull request merged**